src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -380,6 +430,26 @@ public String getRetrySettingsTypeName() {
     return getNotImplementedString("SurfaceNamer.getRetrySettingsClassName");
   }
 
+  /** Type name of the status code */
+  public String getStatusCodeName(Status.Code code) {
+    return inittedConstantName(Name.upperUnderScore(code.toString()));
+  }
+
+  /** The function name to retrieve default client option */
+  public String getDefaultClientOptionFunctionName(Interface service) {
+    return getNotImplementedString("SurfaceNamer.getDefaultClientOptionFunctionName");
+  }
+
+  /** The type name of call options */
+  public String getCallOptionsTypeName(Interface service) {
+    return getNotImplementedString("SurfaceNamer.getCallOptionsTypeName");
+  }
+
+  /** The function name to retrieve default call option */
+  public String getDefaultCallOptionFunctionName(Interface service) {
+    return getNotImplementedString("SurfaceNamer.getDefaultCallOptionFunctionName");
+  }
```

Comment:
Some naming stuff:

* You have singular `Option` in a couple methods above where they should be `Options`
* There are already existing methods to handle API-wrapper-level settings and call-level settings. We should use consistent naming at this level.
 * `getDefaultClientOptionFunctionName`: Maybe `getDefaultApiSettingsFunctionName`
 * `getCallOptionsTypeName`: Maybe `getCallSettingsTypeName`
   * related: rename existing `getSettingsMemberName` to `getCallSettingsMemberName`, and `getSettingsFunctionName` to `getCallSettingsFunctionName`
 * `getDefaultCallOptionFunctionName`: Maybe `getDefaultCallSettingsFunctionName`

Also, prefer to give a default implementation for strings if you can come up with something reasonable.